launcher/spec/launch_policy_test.go

```diff
@@ -114,14 +116,57 @@ func TestVerify(t *testing.T) {
 			true,
 		},
 		{
-			"allow everything",
+			"log redirect violation 1",
 			LaunchPolicy{
-				AllowedEnvOverride: []string{"foo"},
-				AllowedCmdOverride: true,
+				AllowedLogRedirect: never,
+			},
+			LaunchSpec{
+				LogRedirect: true,
+				Hardened:    true,
+			},
+			true,
+		},
+		{
+			"log redirect violation 2",
+			LaunchPolicy{
+				AllowedLogRedirect: never,
+			},
+			LaunchSpec{
+				LogRedirect: true,
```

Comment:
here, we should test LogRedirect: false, Hardened: false. in general, let's test all combinations of `AllowedLogRedirect`, `LogRedirect`, and `Hardened` to ensure we're doing the right thing.

Reply:
added all possible combinations